### PR TITLE
removes ? from URL when deleting all params

### PR DESCRIPTION
**Pull request by justjavac**
wpt: https://github.com/web-platform-tests/wpt/blob/master/url/urlsearchparams-delete.any.js#L32-L45

```ts
const url = new URL('http://example.com/?');
url.href === 'http://example.com/?'
//                               ↑
//                              HERE(with ?)

// Removing non-existent param
url.searchParams.delete('param1'); 
url.href === 'http://example.com/'
//                               ↑
//                              HERE(without ?)
```

### CODE REVIEW

File: js/url.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ function parse(url: string): URLParts | undefined {
 }
 
 export class URL {
-  private _parts: URLParts;
+  _parts: URLParts;
```
Comment (ry) [inline js/url.ts]: Why remove private ?

Comment (justjavac) [reply js/url.ts]: in spec, `URLSearchParams`

> 1. A URLSearchParams object has an associated list of name-value pairs, which is initially empty.
> 1. A URLSearchParams object **has an associated url object**, which is initially null.

and when delete a key, run a [update steps](https://url.spec.whatwg.org/#concept-urlsearchparams-update):

> A `URLSearchParams` object’s **update steps** are to run these steps:
> 
>     1. Let query be the serialization of URLSearchParams object’s list.
>     1. If query is the empty string, then set query to null.
>     1. **Set url object’s url’s query to query**.

In the current code implementation, the `_parts` is "url object’s url", if it is private, we can't access its `query`.

Comment (justjavac) [reply js/url.ts]: I also considered another way (trick), when access `_parts` outside the class `URLSearchParams`, convert it to `any` that has avoided the private limit.

```ts
(this.url as any)._parts.query = query;
```

Comment (kitsonk) [reply js/url.ts]: I would use the `any` hammer over changing the public API surface of an object.  Though it seems like that is a really odd breaking of encapsulation in the standard.  How does JSDom accomplish this?

Comment (justjavac) [reply js/url.ts]: jsdom use [webidl2js](https://github.com/jsdom/webidl2js) to generate a "wrapper class" `URL`, delegating to an "implementation class". the `_url`  only exists  on impls, not on wrappers.

**Pull request merged**